File: ljh/javascript/level1/SortMyWay.js
// 문자열 내 마음대로 정렬하기 (12/18 - 12/24)
export default class SortMyWay{
    solution = (strings, n) => {
        strings = strings.map((str) => str = str.substring(n, n+1) + str);
        strings.sort();
        strings = strings.map((str) => str = str.slice(1));
    
        return strings;
    }
}

File: ljh/java/src/level1/SortMyWay.java
package level1;

import java.util.Arrays;
import java.util.Comparator;

/*
 * 문자열 내 마음대로 정렬하기
 */

public class SortMyWay {
	
    public String[] solution(String[] strings, int n) {
    	int index = n;
    	
        Comparator<String> comp = new Comparator<String>() {
        	@Override
        	public int compare(String o1, String o2) { 
        		int i = 0;
        		if(o1.charAt(index) - o2.charAt(index) != 0) {
        			return o1.charAt(index) - o2.charAt(index);
        		}
        		
        		while(o1.charAt(i)-o2.charAt(i) == 0) {
        			i++;
        		}
        		return o1.charAt(i) - o2.charAt(i);
        	}
        };
        
        Arrays.sort(strings, comp);
        return strings;
    }
}

